frontend/src/App.jsx

```diff
@@ -13,8 +13,113 @@ import WorkspaceFetcher from "@/components/ui/WorkspaceFetcher";
+  const [configuration] = useAtom(activeConfigurationAtom);
+
+  useEffect(() => {
+    const serverAddress = import.meta.env.VITE_EXPRESS;
+    const res = axios.get(`${serverAddress}/isPackaged`).then((response) => {
+      setIsPackaged(response.data);
+      if (response.data === true) {
+        ping(configuration)
+          .then((res) => {
+            if (!res) {
+              axios
+                .get(`${serverAddress}/get-kube-contexts`)
+                .then((res) => {
+                  setAvailableKubeContexts(res.data);
+
+                  axios
+                    .get(`${serverAddress}/get-anvil-config`)
+                    .then((res) => {
+                      const data = res.data;
+                      if (data.has_config == true) {
+                        const config = data.config;
+                        const bucketPort = config.Local.BucketPort;
+                        const driver = config.Local.Driver;
+                        const serverPort = config.ServerPort;
+                        const context = config.KubeContext;
+                        if (config.IsLocal === true) {
+                          setConfirmationIsOpen(true);
+                          const configText = [
+                            "Are you sure you want to run anvil with the following configurations?",
+                            "If you are using minikube driver, please make sure you've setted up your minikube cluster, and that it's running.",
+                            `HOST: 127.0.0.1`,
+                            `PORT: ${serverPort}`,
+                            `Context: ${context}`,
+                            `Driver: ${driver}`,
+                          ];
+                          setConfirmationText(configText);
+                        }
+                      } else {
+                        setConfigOpen(true);
+                        setConfirmationIsOpen(false);
+                      }
+                    })
+                    .catch((err) => {
+                      setErrText([
+                        "Your local anvil did not started as expected. Please select a config",
+                        err.response?.data?.err,
+                        err.response?.data?.kubeErr,
+                      ]);
+                      setConfigOpen(true);
+                    });
+                })
+                .catch((err) => {
+                  setErrText([
+                    "Cannot find kubectl or there's an error with kubectl command. Please try again or check if kubectl is in your path. You can ignore this message if you're using cloud anvil",
+                    err.message,
+                  ]);
+                  setErrModalOpen(true);
+                });
+            } else {
+              //do nothing, because you can ping
+              setConfigOpen(false);
+            }
+          })
+          .catch((err) => {
+            setConfigOpen(true);
+          });
```

Comment:
I agree, but initially when I tried async functions, it was giving me some weird error because I was using async on the top level. But I can take another look, during UX fixes

Reply:
If it's because you can can't make a useEffect callback async, the solution is usually to declare a new function in your useEffect, make it async, then call it at the end of the useEffect callback. https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret